client/app/components/calendar/CalendarRow.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { If } from 'react-if';
+import * as DateHelper from 'helpers/DateHelper';
+import EventSchema from 'schemas/EventSchema';
+
+import EventWrapper from './event/EventWrapper';
+
+const SECONDS_IN_DAY = 24 * 60 * 60;
+
+const { string, bool, number, arrayOf, oneOfType, instanceOf } = React.PropTypes;
+
+export default class CalendarRow extends React.Component {
+  static propTypes = {
+    events:          arrayOf(arrayOf(EventSchema.only('start'))).isRequired,
+    time:            oneOfType([instanceOf(Date), string]).isRequired,
+    days:            arrayOf(oneOfType([instanceOf(Date), string])).isRequired,
+    unitEventLength: number,
+    timeFormat:      string,
+    displayMinutes:  bool
+  };
+
+  static defaultProps = {
+    timeFormat:     'Ha',
+    displayMinutes: false
+  };
+
+  render() {
+    return (
+      <tr>
+        <TimeCell visible={this._displayTime()}
+                  time={this.props.time}
+                  timeFormat={this.props.timeFormat} />
+        {this._tableCellNodes()}
+      </tr>
+    );
+  }
+
+  _eventStartsAt(timestamp) {
+    return (event) => DateHelper.timestamp(event.start.date_time) === timestamp;
+  }
+
+  _eventGroupContaining(timestamp) {
+    return this.props.events.find(group =>
+      group.length &&
+      Math.abs(group[group.length - 1].start_timestamp - timestamp) < SECONDS_IN_DAY &&
+      group.some(event => event.start_timestamp == timestamp)
+    );
+  }
+
+  _eventsStartingAt(timestamp, group) {
+    return group.filter(event => event.start_timestamp == timestamp);
+  }
+
+  _displayTime() {
+    return new Date(this.props.time).getMinutes() === 0 || this.props.displayMinutes;
+  }
+
+  _tableCellNodes() {
+    let currentTimeStamp = DateHelper.timestamp(this.props.days[0], this.props.time) - SECONDS_IN_DAY;
+    return this.props.days.map(() => {
+      currentTimeStamp += SECONDS_IN_DAY;
```

Comment:
I think the point of this was that generating the timestamp every time was very inefficient and was causing the events to load very slowly.